src/Mod/Part/Gui/TaskSweep.cpp

```diff
@@ -277,9 +287,16 @@ bool SweepWidget::accept()
         return false;
     Gui::SelectionFilter edgeFilter  ("SELECT Part::Feature SUBELEMENT Edge COUNT 1..");
     Gui::SelectionFilter partFilter  ("SELECT Part::Feature COUNT 1");
+    Gui::SelectionFilter linkFilter  ("SELECT App::Link COUNT 1");
```

Comment:
Consider replacing `SelectionFilter` here. Just obtain the selected shape and check for the shape type.